pkg/chaosdaemon/iptables_server.go

```diff
@@ -94,10 +94,10 @@ func (iptables *iptablesClient) setIptablesChain(chain *pb.Chain) error {
 	var matchPart string
 	var interfaceMatcher string
 	if chain.Direction == pb.Chain_INPUT {
-		matchPart = "src"
+		matchPart = "src,dst"
```

Comment:
Why do we need this modification? If I'm not wrong, it means this chain applies, iff both the source and destination are in the ipset, which seems not to be what we want?

Reply:
This means it must match a pair of source IP address and destination port. Please refer to [man iptables-extensions - section set, `--match-set` parameter](https://ipset.netfilter.org/iptables-extensions.man.html) and [man ipset - section introduction, set dimensions](https://ipset.netfilter.org/iptables-extensions.man.html).

Reply:
I think we need more documentation to describe it but it still makes sense:

- with `action: delay/loss/...` (which requires `tc`) and `direction: from`, `externalTargets` would not works
- with `action: partition`, `direction: from`, `externalTagets: [10.96.0.0/16:8080]`, if would drop any packet from `10.96.0.0/16` to `<pod-ip>:8080`.